### PR TITLE
Update Intro.lhs

### PR DESCRIPTION
**Pull request by marcinmrotek**
Added some more explanation to the tutorial.

### CODE REVIEW

**Comment by jonsterling** [comment]
Thanks! I will add a few line-comments to the PR

File: tests/Intro.lhs [hunk 1]
```diff
@@ -29,17 +29,25 @@ extensions first:
...
 
-Let’s define a universe of fields which we want to use:
+Let’s define a universe of fields which we want to use.
+
+First of all, we need a data type defining the field labels:
 
 > data Fields = Name | Age | Sleeping | Master deriving Show
+
+Any record can be now described by a type-level list of these labels.
+The DataKinds extension must be enabled to autmatically turn all the constructors of the Field type into types.
```
Comment (jonsterling) [inline tests/Intro.lhs]: literate haskell requires one line of whitespace between code and text

File: tests/Intro.lhs [hunk 2]
```diff
@@ -29,17 +29,25 @@ extensions first:
...
 > type LifeForm = [Name, Age, Sleeping]
 
+Now, we need a way to map our labels to concrete types. We use a type family for this purpose:
```
Comment (jonsterling) [inline tests/Intro.lhs]: likewise, add a line of whitespace here

File: tests/Intro.lhs [hunk 3]
```diff
@@ -29,17 +29,25 @@ extensions first:
...
 > type family ElF (f :: Fields) :: * where
 >   ElF Name = String
 >   ElF Age = Int
 >   ElF Sleeping = Bool
 >   ElF Master = Rec Attr LifeForm
 
+Unfortunately, type families aren't first class in Haskell.
+That's why we also need a data type, with which we will parametrise Rec:
```
Comment (jonsterling) [inline tests/Intro.lhs]: likewise, add a line of whitespace here

File: tests/Intro.lhs [hunk 4]
```diff
@@ -47,9 +55,14 @@ Let’s define a universe of fields which we want to use:
 > instance Show (Attr Sleeping) where show (Attr x) = "sleeping: " ++ show x
 > instance Show (Attr Master) where show (Attr x) = "master: " ++ show x
 
+To make field construction easier, we define an operator.
+The first argument of this operator is a singleton - a constructor bringing the data-kinded field label type into the data level.
+It's needed because there can be multiple label with the same field type,
+so by just supplying a value of type (ElF f) there would be no way to deduce the correct "f".
```
Comment (jonsterling) [inline tests/Intro.lhs]: - likewise, add a line of whitespace here

File: tests/Intro.lhs [hunk 5]
```diff
@@ -47,9 +55,14 @@ Let’s define a universe of fields which we want to use:
...
 > (=::) :: sing f -> ElF f -> Attr f
 > _ =:: x = Attr x
 
+We generate the necessary singletons for each field label using Template Haskell:
```
Comment (jonsterling) [inline tests/Intro.lhs]: likewise, add a line of whitespace here

**Comment by marcinmrotek** [comment]
Ah, sorry, I've never written LH before, and I just edited this file
through Github. Thanks for fixing it ;)

**Comment by slpopejoy** [comment]
Can I pile in on this and add that the tutorial should specify that the singletons package is required?

Ie, change `cabal install vinyl` to `cabal install vinyl singletons`

**Comment by acowley** [comment]
I pulled the commit in with the whitespace corrections and @slpopejoy's recent tweak, but had to do a bit of branch surgery to straighten things out.

**Pull request closed without merging**